### PR TITLE
Star Trek episode writers tutorial

### PR DESCRIPTION
**Pull request by ronichoudhury**
I realized we already have a "developer cookbook" in the docs, in the Tutorials section.  This PR adds an example using CSV data to create a SQLite database with SQLAlchemy, then builds a web service and a frontend to visualize some of the data.

@jeffbaumes please take a look.  I think some parts of this may not make sense as written, or as ordered.  Let me know what you think.

FYI: This PR also bumps up the required Sphinx version (for development only).

### CODE REVIEW

File: docs/tutorials/db-vis.rst [hunk 1]
```diff
@@ -0,0 +1,360 @@
...
+indicating its overall episode number, season/episode, title, airdate, a link to
+the associated Memory Alpha article, and numeric indices into *people.csv* to
+indicate who wrote each teleplay, who developed each story, and who directed
+each episode.
+
+Creating the Database
+=====================
+
+SQLAlchemy is a Python library that provides a programmatic API for creating,
+updating, and accessing SQLite databases.  It includes an *object-relational
```
Comment (jeffbaumes) [inline docs/tutorials/db-vis.rst]: Perhaps change "SQLite" to "SQL" to convey SQLAlchemy's generality.

File: docs/tutorials/db-vis.rst [hunk 2]
```diff
@@ -0,0 +1,360 @@
...
+mapping (ORM)* component, meaning it provides facilities for writing Python
+classes that transparently maintain a connection to the database, changing it as
+the object is updated, etc.
+
```
Comment (jeffbaumes) [inline docs/tutorials/db-vis.rst]: Should add text here describing how to install sqlalchemy with pip.

File: docs/tutorials/db-vis.rst [hunk 3]
```diff
@@ -0,0 +1,360 @@
...
+objects, one per data point.  As a starting point for a visualization project on
+Star Trek episode data, let's tally up the number of episodes written or
+developed by each person in the *people* table, and use Vega to render a bar
+chart.  To do so, we need to query the database and count how many episodes each
+person is associated to.  We can use the ORM classes to accomplish this.  Let's
+analyze the file `writers.py <../_static/tng/writers.py>`_ to see how.  First,
+module imports:
+
+.. literalinclude:: ../static/tng/writers.py
+    :lines: 1-4
```
Comment (jeffbaumes) [inline docs/tutorials/db-vis.rst]: You need to include line 5 here for the Episodes import.

File: docs/tutorials/db-vis.rst [hunk 4]
```diff
@@ -0,0 +1,360 @@
...
+
+With this file written we have the start of a web application.  To see how
+things stand, you can launch Tangelo to serve this directory to the web,
+
+.. code-block:: shell
+
+    tangelo --root .
+
+and then visit http://localhost:8080/writers to see the list of JSON objects
+that results.
```
Comment (jeffbaumes) [inline docs/tutorials/db-vis.rst]: This does not seem to work without the `?sort=true` (or `false`). Without it I get the error:
```
{"error": "'sort' was registered for type conversion but did not appear in the arguments list"}
```

File: docs/tutorials/db-vis.rst [hunk 5]
```diff
@@ -0,0 +1,360 @@
...
+simple.  Here is the webpage itself, in `index.html <../_static/tng/index.html>`_:
+
+.. literalinclude:: ../static/tng/index.html
+    :language: html
+    :linenos:
+
+This is a very simple HTML file with a ``div`` element (line 9), in which we
+will place a Vega visualization.
+
+Next, we have some simple JavaScript to go along with this HTML file:
```
Comment (jeffbaumes) [inline docs/tutorials/db-vis.rst]: Should add a link to index.js file like you have done for all other files.

**Comment by jeffbaumes** [comment]
See my inline comments. With those changes someone should be able to do the example start to finish by building up the files themselves through cut/paste code from the tutorial. The only thing I downloaded directly were the two data files. Nice tutorial!

**Comment by ronichoudhury** [comment]
@jeffbaumes, thanks for your detailed comments!  I have made the recommended changes in e1d8ecae5c702ba244cd5b8c1bdd72cec33f07ae.

I'm also glad to know that the tutorial works, even via cut and paste!

Let me know if the changes are good and I will go ahead and merge.

**Comment by jeffbaumes** [comment]
Changes are good, thanks. LGTM :checkered_flag:

**Pull request merged**